include/cantera/thermo/ThermoPhase.h

```diff
@@ -1709,6 +1709,20 @@ class ThermoPhase : public Phase
     virtual void setParameters(const AnyMap& phaseNode,
                                const AnyMap& rootNode=AnyMap());
 
+    //! Store the parameters of a ThermoPhase object such that an identical
+    //! one could be reconstructed using the newPhase(AnyMap&) function. This
+    //! does not include user-defined fields available in input().
+    virtual void getParameters(AnyMap& phaseNode) const;
+
+    //! Get phase-specific parameters of a Species object such that an
+    //! identical one could be reconstructed and added to this phase.
+    /*!
+     * @param name     Name of the species
+     * @param eosNode  Mapping to be populated with parameters
+     */
+    virtual void getSpeciesParameters(const std::string& name,
```

Comment:
Is a dedicated `getSpeciesParameters` necessary? Presumably parameters can be accessed via `species[name].getParameters(eosNode)` or similar. The implementation seems like an outlier, as there is no corresponding `getReactionParameters` in `Kinetics.h`.

Reply:
The reason for this method is that the behavior of `getSpeciesParameters` is specialized for different `ThermoPhase` models, and it is collecting data stored in the `ThermoPhase` object that applies to the specified species. Most of the time, this is really just populating the `equation-of-state` child node, but I think there were a couple of cases where there were fields outside that that needed to be set as well.

This is somewhat different from the case of reactions, where much of the specialized behavior is handled by the different `Reaction` classes, while the `GasKinetics` class is mostly generic (and getting more generic thanks to your work on #995).

Reply:
Makes sense.